### ISSUE TITLE
LinkedIn deprecated OAuth login in favour of OIDC

### ISSUE THREAD

**Issue by yktoo** [markbates/goth#519]
The [deprecation notice](https://learn.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin):

> This version of Sign In with LinkedIn has been deprecated as of August 1, 2023. For all Sign In with LinkedIn implementations going forward, please refer to [Sign In with LinkedIn using OpenID Connect](https://learn.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin-v2).

This leads to a change of scopes: instead of `r_liteprofile`, `r_emailaddress` only the following is now available:

* `oidc`
* `profile`
* `email`

As a consequence, fetching profile in goth now fails with **403 Forbidden** ([here](https://github.com/markbates/goth/blob/492927988958a49753893bce51f9c972dce82929/providers/linkedin/linkedin.go#L116)).

**Comment by techknowlogick** [comment]
Thanks for reporting this @yktoo

In this case, would changing the scopes in the code be sufficient, or would any API endpoints need to be updated?

**Comment by yktoo** [comment]
No, I tried with the above scopes and that's what results in a 403. I'm not sure what else needs to be changed.

**Comment by trendamplifier** [comment]
I have this issue too, can verify what @yktoo said. Please advise.

**Comment by yktoo** [comment]
I managed to successfully add LinkedIn login using the standard OIDC provider. In the end it turned out to be not that much different from other `goth` providers.

You may want to have a look at Comentario [auth implementation](https://gitlab.com/comentario/comentario/-/blob/master/internal/config/oauth.go?ref_type=heads#L156) and [docs](https://docs.comentario.app/en/configuration/idps/linkedin/).

I believe the issue can be closed now.
